README.md

```diff
@@ -0,0 +1,44 @@
+# Firmware for Signet Devices
+
+## Prerequisites
+
+#### Cloning Directions
+
+This project uses git submodules, you will either need to use `git clone
+--recursive <repourl>` or after cloning use `git submodule update --init`.
+
+#### Dependencies
+
+The following packages must be installed:
+
+`apt install build-essential autoconf automake gperf texinfo help2man ncurses-dev bison flex libtool-bin libjson-c-dev libiberty-dev`
+
+## Building Firmwares
+
+#### Prepare the Toolchain, Dependencies and Tools
+
+For the legacy Signet go to `firmware` directory, for Signet High-capacity go
+to `firmware-hc`, and run once `./build-deps.sh` to build the corresponding
+toolchain and dependencies.
+
+For packing the firmware in a suitable format for the Signet client GUI we will
+need a firmware encoder too. `make signet-firmware-encoder` for Signet and
+`make hc-firmware-encoder` for Signet High-capacity.
+
+#### Build the Firmware
+
+`make` builds the Signet firmware, while Signet HC needs two separate commands
+`make BT_MODE=A` and `make BT_MODE=B`.
+
+Packaging Signet or Signet HC firmware, respectively:
+```
+./signet-firmware-encoder signet-fw.bin signet-fw.sfw
+./hc-firmware-encoder signet-fw-a.bin signet-fw-b.bin signet-fw-0.2.3.sfwhc 0 2 3
+```
+
+#### Reproducible Builds
+
+Both Signet and Signet High-capacity firmware uses locally compiled toolchains
+(via crosstool-ng) to achieve reproducible builds. You can verify that
+compiling the source code results in identical firmware binaries to the ones
+downloadable from https://nthdimtech.com.
```

Comment:
Oh, ye of little faith. :)
Sure, your choice. Changed.

It would be interesting to compile it on MacOS if you happen to have one.
./hc-firmware-encoder signet-fw-a.bin signet-fw-b.bin signet-fw-0.2.2.sfwhc 0 2 2
sha256sum signet-fw-0.2.2.sfwhc
2311e61047a8f75a3242e0392de63de66488d4e8d9ff1c2fa6dc91293fadafed
In three different CPU/OS combinations. Pretty promising, I'd say!